tests/Makefile.am

```diff
@@ -115,20 +116,25 @@ v4l2decode_LDADD += -ldl
 
 
 v4l2encode_LDADD   = $(V4L2_ENCODE_LIBS)
-v4l2encode_LDFLAGS = $(V4L2_ENCODE_LDFLAGS)
+v4l2encode_CPPFLAGS = $(YAMI_COMMON_CFLAGS)
+v4l2encode_LDFLAGS = $(V4L2_ENCODE_LDFLAGS) $(AM_LDFLAGS)
 v4l2encode_SOURCES = v4l2encode.cpp encodeinput.h encodeinput.cpp encodeInputCamera.cpp encodeInputDecoder.cpp $(DECODE_INPUT_SOURCES)
 
 yamivpp_LDADD    = $(YAMI_VPP_LIBS)
-yamivpp_LDFLAGS  = $(YAMI_VPP_LDFLAGS)
+yamivpp_CPPFLAGS = $(YAMI_COMMON_CFLAGS)
+yamivpp_LDFLAGS  = $(YAMI_VPP_LDFLAGS) $(AM_LDFLAGS)
 yamivpp_SOURCES  = vppinputdecode.cpp vppinputoutput.cpp vppoutputencode.cpp  vpp.cpp encodeinput.cpp encodeInputCamera.cpp encodeInputDecoder.cpp $(DECODE_INPUT_SOURCES) vppinputdecodecapi.cpp
 
 yamitranscode_LDADD    = $(YAMI_VPP_LIBS)
-yamitranscode_LDFLAGS  = -pthread $(YAMI_VPP_LDFLAGS)
+yamitranscode_CPPFLAGS = $(YAMI_COMMON_CFLAGS)
+yamitranscode_LDFLAGS  = -pthread $(YAMI_VPP_LDFLAGS) $(AM_LDFLAGS)
 yamitranscode_SOURCES  = vppinputdecode.cpp vppinputoutput.cpp vppoutputencode.cpp  yamitranscode.cpp encodeinput.cpp encodeInputCamera.cpp encodeInputDecoder.cpp $(DECODE_INPUT_SOURCES) vppinputasync.cpp vppinputdecodecapi.cpp 
 
 bin_PROGRAMS += yamiinfo
 yamiinfo_SOURCES = yamiinfo.cpp
+yamiinfo_CPPFLAGS = $(YAMI_COMMON_CFLAGS)
 yamiinfo_LDADD = $(YAMI_COMMON_LIBS)
 yamiinfo_LDFLAGS = -Wl,--no-as-needed \
 	$(LIBYAMI_CFLAGS) \
```

Comment:
`$(LIBYAMI_CFLAGS)` should not belong in `yamiinfo_LDFLAGS`.  Note: you've already included it in `$(YAMI_COMMON_CFLAGS)`